apps/transport/lib/db/dataset.ex

```diff
@@ -278,24 +274,23 @@ defmodule DB.Dataset do
 
   @spec filter_by_aom(Ecto.Query.t(), map()) :: Ecto.Query.t()
   defp filter_by_aom(query, %{"aom" => aom_id}) do
-    case parent_dataset(aom_id) do
-      nil -> where(query, [d], d.aom_id == ^aom_id)
-      parent_dataset_id -> where(query, [d], d.aom_id == ^aom_id or d.id == ^parent_dataset_id)
-    end
-  end
-
-  defp filter_by_aom(query, _), do: query
+    aom_id =
```

Comment:
Je propose cette version alternative de la fonction :
```elixir
  defp filter_by_aom(query, %{"aom" => aom_id}) do
    aom_id = String.to_integer(aom_id)
    query
    |> join(:left, [dataset: d], aom in assoc(d, :legal_owners_aom), on: aom.id == ^aom_id, as: :aom)
    |> where([dataset: d, aom: aom], d.aom_id == ^aom_id or aom.id == ^aom_id)
  end
```

Explications:
* si `aom_id` vaut nil après avoir été parsée par `Integer.parse`, on va avoir une condition `where: a.aom_id == nil,`, ce qui est interdit avec Ecto.
![image](https://github.com/etalab/transport-site/assets/15341118/d78a7245-cf57-4587-b3f4-2efeb50ede06)
* il n'est normalement jamais nécessaire de faire appel aux tables intermédiaires des relations many-to-many. Voir la doc à ce [sujet](https://hexdocs.pm/ecto/Ecto.Schema.html#many_to_many/3-using-ecto-assoc-2). "One of the benefits of using many_to_many is that Ecto will avoid loading the intermediate whenever possible, making your queries more efficient. For this reason, developers should not refer to the join table of many_to_many in queries."
* `String.to_integer(aom_id)` va crasher en cas de problème, mais ça ne pose de problème car le cas est déjà géré ailleurs.
![image](https://github.com/etalab/transport-site/assets/15341118/1e357c80-4d1d-4aba-809c-74edd1150f9c)